src/bounds/ellipsoid.jl

```diff
@@ -20,8 +20,9 @@ end
 
 
 function Ellipsoid(center::AbstractVector, A::AbstractMatrix)
-    axes, axlens = decompose(A)
-    Ellipsoid(center, A, axes, axlens, _volume(A))
```

Comment:
Rather than incorporate this here, what if we wrote
```julia
decompose(A::AbstractMatrix) = decompose(Symmetric(A))

function decompose(A::SymmetricMatrix)
	E = eigen(A)
	#...
end
```

Reply:
I think it makes most sense to have `Ellipsoid.A` field to have type of `Symmetric`: covariance matrices are symmetric anyway. Unfortunately, it doesn't work because inplace broadcasting with `Symmetric` isn't implemented in julia yet - there are several issues, eg https://github.com/JuliaLang/julia/issues/38056. So for now, in this PR, I put the `Symmetric` wrapper as "high" in the call stack as possible. Shifting it to the `decompose()` level is also possible, of course. Would you prefer that variant?

Reply:
Thanks for pointing out the mutation issues- I wasn't aware of those. Yes, I prefer to have this at the decompose level, it's not really too different from what you wrote, but I like using the method signatures. If we had a trait system, we could easily switch from `decompose(::Symmetric)` to something like `decompose(::T) where issymmetric(T)`. FYI, I do have hopes of improving the entire `Bounds` interface, since each of these bounds is a sample-able (i.e., implement the `Distributions.jl` API for samplers), which would also make them immutable.

Reply:
Ok, changed.
My point was that it mathematically only makes sense to run `decompose` with `Symmetric` matrices, same for the `Ellipsoid` creation and other related functions. Shouldn't need any traits even potentially, I think.

Reply:
Good point, I've noted that in my ideas for the improvements I alluded to.